Prepare 3.54.0 (#3112)
SFDO-Tooling/CumulusCI: HISTORY.rst

History
=======

3.54.0 (2022-03-10)
-------------------

Changes

> - We added the ``apex_compile_type`` and ``upgrade_type`` options for performing Unlocked Package installations. (#3105)

Reviewer: ```suggestion
- The ``install_managed`` and ``update_dependencies`` tasks now support ``apex_compile_type`` and ``upgrade_type`` options for performing Unlocked Package installations. (#3105)
```

> - We added support for new Slack metadata types (#3103)

Reviewer: ```suggestion
- The ``update_package_xml`` task now supports new Slack metadata types (#3103)
```

- We added the ability to do Upserts during Data Loading (`documentation <https://cumulusci.readthedocs.io/en/stable/data.html#upserts>`__). (#3099)

> - Robot tests can now pass a locator when calling ``Go To Page`` for one of our bulit-in page objects (Listing, Detail, Home), and the keyword will wait until that locator is visible. This is mostly useful for custom pages where you need to wait for an iframe or custom component to be available. (#3098)

Reviewer: ```suggestion
- Robot Framework tests can now pass a locator when calling ``Go To Page`` for one of our bulit-in page objects (Listing, Detail, Home), and the keyword will wait until that locator is visible. This is mostly useful for custom pages where you need to wait for an iframe or custom component to be available. (#3098)
```

> - The ``update_dependencies`` and ``install_managed`` tasks accept the options ``interactive`` and ``base-package-url-format``, which pause the automation and wait for the user to confirm the list of packages to be installed. (#3093)

Reviewer: Move this up next to the other release note about these tasks?

Author: ☹️ but they're in PR number order

Reviewer: I think it's more helpful to readers to group them thematically when there's an obvious way to do that, rather than by the order in which they were submitted, which shouldn't really matter to our users. But it's not that important.

Author: How's this?
```rst
- The ``install_managed`` and ``update_dependencies`` tasks:
  - Now support ``apex_compile_type`` and ``upgrade_type`` options for performing Unlocked Package installations. (#3105)
  - Accept the options ``interactive`` and ``base-package-url-format``, which pause the automation and wait for the user to confirm the list of packages to be installed. (#3093)
```

Author: > ☹️ but they're in PR number order

Sorry, I should have been clearer that this was a joke.
